src/game_events/manager_impl.cpp

```diff
@@ -94,70 +93,51 @@ handler_list& event_handlers::get(const std::string& name)
  * An event with a nonempty ID will not be added if an event with that
  * ID already exists.
  */
-void event_handlers::add_event_handler(const config& cfg, bool is_menu_item)
+pending_event_handler event_handlers::add_event_handler(const std::string& name, const std::string& id, bool repeat, bool is_menu_item)
 {
-	// Someone decided to register an empty event... bail.
-	if(cfg.empty()) {
-		return;
-	}
-
-	std::string name = cfg["name"];
-	std::string id = cfg["id"];
-
 	if(!id.empty()) {
 		// Ignore this handler if there is already one with this ID.
 		auto find_it = id_map_.find(id);
 
 		if(find_it != id_map_.end() && !find_it->second.expired()) {
 			DBG_EH << "ignoring event handler for name='" << name << "' with id '" << id << "'\n";
-			return;
+			return {*this, nullptr};
 		}
 	}
 
 	if(name.empty() && id.empty()) {
 		lg::log_to_chat() << "[event] is missing name or id field\n";
```

Comment:
Why remove the debugging here?

This looks like an incorrectly resolved merge-conflict with #6619's bfd0ef9af7aace4d753521fb29e9ad7b9c9419c7.

Reply:
If this is about the code that dumps the entire content of the event, it's not removed. The event content is no longer available in this function, so that debug message was moved into the caller.

Reply:
I don't see where it moved to, and the logging seems to be missing when retesting #6619.

Steps:
1. In EI S01, add `{EXTRA_SCENARIO_MUSIC nunc_dimittis.ogg}` immediately after the opening `[scenario]` tag.
2. Run Wesnoth with `--log-debug=wml`, and start the scenario.
3. The "[event] is missing name or id field" is shown in chat
4. No additional log is printed, neither on the Lua console nor the command-line console.

Reply:
The message is there and works as expected, with one minor caveat. It moved to the `event_handler` logdomain. Is that good enough? Should I change the logdomain for that one message?

Reply:
It needs to be useful for debugging - in the old code it had a message printed to the console at log level error or warning, with additional information printed immediately after that if used with `--log-debug=wml`.

With the new code, using `--log-warning=event_handler` doesn't print anything. Using `--log-debug=event_handler` prints 673 lines, with these looking very in-place among them; this doesn't even look like an error message:
```
Checking lua scripts... ok
20220704 00:55:38 debug event_handler: Content of that event:
[music]
	append = yes
	name = nunc_dimittis.ogg
[/music]

20220704 00:55:38 debug event_handler: Registered WML event 'prestart' (first time only) with the following actions:
```

Reply:
Okay, so the correct fix is:

1. Move it to the WML logdomain?
2. Print the error to the console in the event_handler logdomain?
3. Both, in the WML logdomain?

Reply:
> Okay, so the correct fix is: ... ?

I know that I push for "more documentation" often, but in this case I'd say the correctness of the fix should be judged by writing a reply to https://r.wesnoth.org/p672974 that starts "From 1.17.6, use ...". Either of these sound good:

- use `--log-debug=wml`, just as before.
- use `--log-info=event_handler` instead.

Given the amount of output from `--log-debug=event_handler`, I think the current message has to either move to the `wml` logdomain or be promoted to `info` severity in the `event_handler` logdomain. In either case, print the one-line error to the console in the same logdomain too.

Reply:
Thanks for resolving.